File: generated-code/example/javascript/model/one-of.js
/**
 * Oneof example
 */
class OneOf {
    /**
     * Read OneOf from input stream
     */
    static async readFrom(stream) {
        let tag = await stream.readInt();
        if (tag == OptionOne.TAG) {
            return await OptionOne.readFrom(stream);
        }
        if (tag == OptionTwo.TAG) {
            return await OptionTwo.readFrom(stream);
        }
        throw new Error("Unexpected tag value");
    }
}
/**
 * First option
 */
class OptionOne extends OneOf {
    /**
     * List of integers
     */
    vecInt;
    /**
     * Long integer
     */
    longInt;

    constructor(vecInt, longInt) {
        super();
        this.vecInt = vecInt;
        this.longInt = longInt;
    }

    /**
     * Read OptionOne from input stream
     */
    static async readFrom(stream) {
        let vecInt;
        vecInt = [];
        for (let vecIntCount = await stream.readInt(); vecIntCount > 0; vecIntCount--) {
            let vecIntElement;
            vecIntElement = await stream.readInt();
            vecInt.push(vecIntElement);
        }
        let longInt;
        longInt = await stream.readLong();
        return new OptionOne(vecInt, longInt);
    }

    /**
     * Write OptionOne to output stream
     */
    async writeTo(stream) {
        await stream.writeInt(OptionOne.TAG);
        let vecInt = this.vecInt;
        await stream.writeInt(vecInt.length);
        for (let vecIntElement of vecInt) {
            await stream.writeInt(vecIntElement);
        }
        let longInt = this.longInt;
        await stream.writeLong(longInt);
    }
}

OptionOne.TAG = 0;
OneOf.OptionOne = OptionOne;
/**
 * Second option
 */
class OptionTwo extends OneOf {
    /**
     * usize
     */
    value;

    constructor(value) {
        super();
        this.value = value;
    }

    /**
     * Read OptionTwo from input stream
     */
    static async readFrom(stream) {
        let value;
        value = await stream.readInt();
        return new OptionTwo(value);
    }

    /**
     * Write OptionTwo to output stream
     */
    async writeTo(stream) {
        await stream.writeInt(OptionTwo.TAG);
        let value = this.value;
        await stream.writeInt(value);
    }
}

OptionTwo.TAG = 1;
OneOf.OptionTwo = OptionTwo;
module.exports = OneOf;

File: generated-code/example/python/model/one_of.py
class OneOf:
    @staticmethod
    def read_from(stream):
        tag = stream.read_int()
        if tag == OptionOne.TAG:
            return OneOf.OptionOne.read_from(stream)
        if tag == OptionTwo.TAG:
            return OneOf.OptionTwo.read_from(stream)
        raise Exception("Unexpected tag value")

class OptionOne(OneOf):
    TAG = 0

    def __init__(self, vec_int, long_int):
        self.vec_int = vec_int
        self.long_int = long_int

    @staticmethod
    def read_from(stream):
        vec_int = []
        for _ in range(stream.read_int()):
            vec_int_element = stream.read_int()
            vec_int.append(vec_int_element)
        long_int = stream.read_long()
        return OptionOne(vec_int, long_int)

    def write_to(self, stream):
        stream.write_int(self.TAG)
        stream.write_int(len(self.vec_int))
        for element in self.vec_int:
            stream.write_int(element)
        stream.write_long(self.long_int)

    def __repr__(self):
        return "OptionOne(" + \
            repr(self.vec_int) + \
            ", " + \
            repr(self.long_int) + \
            ")"

OneOf.OptionOne = OptionOne

class OptionTwo(OneOf):
    TAG = 1

    def __init__(self, value):
        self.value = value

    @staticmethod
    def read_from(stream):
        value = stream.read_int()
        return OptionTwo(value)

    def write_to(self, stream):
        stream.write_int(self.TAG)
        stream.write_int(self.value)

    def __repr__(self):
        return "OptionTwo(" + \
            repr(self.value) + \
            ")"

OneOf.OptionTwo = OptionTwo